sdk/schemaregistry/schema-registry-avro/src/utils/conversion.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { blobToUint8Array } from "./blobToUint8Array";
+
+/**
+ * @param input - Input to `deserialize`.
+ * @returns Promise which completes with the input data as a Uint8Array.
+ */
+export async function toUint8Array(
+  input: Uint8Array | Buffer | Blob | ReadableStream | NodeJS.ReadableStream
+): Promise<Uint8Array> {
+  if (isNodeJSReadableStream(input)) {
+    return streamToBuffer(input);
+  } else if (isWHATWGReadableStream(input)) {
+    return readWHATWGStream(input);
+  } else if ((typeof Blob === "function" || typeof Blob === "object") && input instanceof Blob) {
+    return blobToUint8Array(input);
+  } else return input as Uint8Array;
+}
+
+function isWHATWGReadableStream(input: any): input is ReadableStream {
+  return input && typeof input.pipeThrough === "function";
+}
+
+function isNodeJSReadableStream(input: any): input is NodeJS.ReadableStream {
+  return input && typeof input.pipe === "function";
+}
+
+async function streamToBuffer(input: {
+  [Symbol.asyncIterator](): AsyncIterableIterator<any>;
+}): Promise<Buffer> {
+  const buffers: Buffer[] = [];
+  for await (const chunk of input) {
+    if (Buffer.isBuffer(chunk)) {
+      buffers.push(chunk);
+    } else {
+      buffers.push(Buffer.from(chunk));
+    }
+  }
+  return Buffer.concat(buffers);
+}
+
+// WHATWG's ReadableStream spec supports async iteration: https://streams.spec.whatwg.org/#rs-asynciterator
+// However, Chromium and TypeScript did not implement the support yet
+// see https://bugs.chromium.org/p/chromium/issues/detail?id=929585&q=readablestream%20async&can=2
+// and https://github.com/microsoft/TypeScript/issues/29867
+async function readWHATWGStream(stream: ReadableStream): Promise<Buffer> {
```

Comment:
Am I wrong to think that reading the stream into a buffer defeats the purpose of an asynchronous stream? The user could read the stream to a buffer themselves, but one reason they might pass a stream instead of a buffer would be to avoid allocating memory for the entire file.

(aside, doesn't this also require us to polyfill buffer?)

Reply:
Yeah you're right, it feels iffy. The whole motivation here is to be consistent with other languages who do the same thing. We're constrained by the 3rd party serializer capabilities at the moment, they only support buffers and strings.

And yes this package depends on the buffer polyfill before this change.

Reply:
I'd be in favor of adding support for whatwg streams if there's core support for it, then packages wouldn't have to explicitly support it, it'd just be folded into `RequestBodyType`.

Reply:
@witemple-msft Yeah that sounds good to me and I will close this for now.